py_ts_interfaces/parser.py

```diff
@@ -80,12 +85,12 @@ def flush(self) -> str:
         return "\n\n".join(serialized).strip()
 
 
-def get_types_from_classdef(node: astroid.ClassDef) -> Dict[str, str]:
+def get_types_from_classdef(node: astroid.ClassDef, classdefs: Dict = {}) -> Dict[str, str]:
```

Comment:
Two concerns here:
1. `Dict` is a generic type in Python, so we can't leave it as just `Dict`.  I think what you want here is `Dict[str, astroid.ClassDef]`
2. It's not good practice to use `{}` as the default value in function parameters.  It doesn't instantiate a new one on every function call, so we could get weird bugs if we leave it this way.

I suggest we do something like this pattern:
```
def get_types_from_classdef(..., classdefs: Optional[Dict[str, astroid.ClassDef]] = None) -> ...:
    if classdefs is None:
        classdefs = {}
    assert classdefs is not None
    ...

Reply:
That's some great insight, I really appreciate the explanation to the better option.
I'll go ahead and change the function parameters to prevent using the same instances of `{}`, and implement the typing for `Dict[str, astroid.ClassDef]`.

Thanks!